In [157]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

try:
    # Attempt to read the CSV with ISO-8859-1 encoding
    df = pd.read_csv('working-df/AviationData.csv', encoding='latin-1')
    print("Successfully read CSV with ISO-8859-1 encoding.")
except UnicodeDecodeError:
    try:
        # If ISO-8859-1 fails, attempt with cp1252 encoding
        df = pd.read_csv('working-df/AviationData.csv', encoding='cp1252')
        print("Successfully read CSV with cp1252 encoding.")
    except UnicodeDecodeError:
         try:
            #If cp1252 fails, attempt with utf-16 encoding
            df = pd.read_csv('working-df/AviationData.csv', encoding='utf-16')
            print("Successfully read CSV with utf-16 encoding.")
         except UnicodeDecodeError as e:
            print(f"Failed to read the CSV with ISO-8859-1, cp1252, and utf-16 encodings. Error: {e}")
            raise # Raise the exception so the user can see the error if none of the encodings work
df.info()

<ipython-input-157-e7125e57eb9f>:8: DtypeWarning: Columns (6,7,28) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('working-df/AviationData.csv', encoding='latin-1')


Successfully read CSV with ISO-8859-1 encoding.
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 88889 entries, 0 to 88888
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      88889 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50132 non-null  object 
 9   Airport.Name            52704 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87507 non-null  obj

In [158]:
df.head()

Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country   Latitude  Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States        NaN        NaN          NaN   
1   BRIDGEPORT, CA  United States        NaN        NaN          NaN   
2    Saltville, VA  United States  36.922223 -81.878056          NaN   
3       EUREKA, CA  United States        NaN        NaN          NaN   
4       Canton, OH  United States        NaN        NaN          NaN   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1          NaN  ...          Personal         NaN                  4.0   
2          NaN  ...          Personal         NaN                  3.0   
3          NaN  ...          Personal         NaN                  2.0   
4          NaN  ...          Personal         NaN                  1.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   
3                    0.0                  0.0             0.0   
4                    2.0                  NaN             0.0   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               UNK                 Cruise  Probable Cause              NaN  
1               UNK                Unknown  Probable Cause       19-09-1996  
2               IMC                 Cruise  Probable Cause       26-02-2007  
3               IMC                 Cruise  Probable Cause       12-09-2000  
4               VMC               Approach  Probable Cause       16-04-1980  

[5 rows x 31 columns]

# Cleaning the Data
rename columns to remove dots as they may cause errors in Python (replace dots with dashes or underscores)

In [159]:
# replace dots in column names with underscores
df.columns = df.columns.str.replace('.', '_')

In [160]:
df.head()

Event_Id Investigation_Type Accident_Number  Event_Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country   Latitude  Longitude Airport_Code  \
0  MOOSE CREEK, ID  United States        NaN        NaN          NaN   
1   BRIDGEPORT, CA  United States        NaN        NaN          NaN   
2    Saltville, VA  United States  36.922223 -81.878056          NaN   
3       EUREKA, CA  United States        NaN        NaN          NaN   
4       Canton, OH  United States        NaN        NaN          NaN   

  Airport_Name  ... Purpose_of_flight Air_carrier Total_Fatal_Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1          NaN  ...          Personal         NaN                  4.0   
2          NaN  ...          Personal         NaN                  3.0   
3          NaN  ...          Personal         NaN                  2.0   
4          NaN  ...          Personal         NaN                  1.0   

  Total_Serious_Injuries Total_Minor_Injuries Total_Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   
3                    0.0                  0.0             0.0   
4                    2.0                  NaN             0.0   

  Weather_Condition  Broad_phase_of_flight   Report_Status Publication_Date  
0               UNK                 Cruise  Probable Cause              NaN  
1               UNK                Unknown  Probable Cause       19-09-1996  
2               IMC                 Cruise  Probable Cause       26-02-2007  
3               IMC                 Cruise  Probable Cause       12-09-2000  
4               VMC               Approach  Probable Cause       16-04-1980  

[5 rows x 31 columns]

Check for duplicate rows


In [161]:
# check for duplicate rows in the Event_Id column
duplicate_rows_events = df[df.duplicated(subset=['Event_Id'], keep=False)]

duplicate_rows_events.head(10)

Event_Id Investigation_Type Accident_Number  Event_Date  \
117  20020917X01908           Accident     DCA82AA012B  1982-01-19   
118  20020917X01908           Accident     DCA82AA012A  1982-01-19   
153  20020917X02259           Accident     LAX82FA049A  1982-01-23   
158  20020917X02400           Accident     MIA82FA038B  1982-01-23   
159  20020917X02400           Accident     MIA82FA038A  1982-01-23   
160  20020917X02259           Accident     LAX82FA049B  1982-01-23   
242  20020917X02585           Accident     SEA82DA028A  1982-02-06   
244  20020917X02173           Accident     LAX82DA065B  1982-02-06   
245  20020917X02585           Accident     SEA82DA028B  1982-02-06   
248  20020917X02173           Accident     LAX82DA065A  1982-02-06   

               Location        Country Latitude Longitude Airport_Code  \
117        ROCKPORT, TX  United States      NaN       NaN          RKP   
118        ROCKPORT, TX  United States      NaN       NaN          RKP   
153     VICTORVILLE, CA  United States      NaN       NaN          NaN   
158  NEWPORT RICHEY, FL  United States      NaN       NaN          NaN   
159  NEWPORT RICHEY, FL  United States      NaN       NaN          NaN   
160     VICTORVILLE, CA  United States      NaN       NaN          NaN   
242         MEDFORD, OR  United States      NaN       NaN          MFR   
244        SAN JOSE, CA  United States      NaN       NaN          RHV   
245         MEDFORD, OR  United States      NaN       NaN          MFR   
248        SAN JOSE, CA  United States      NaN       NaN          RHV   

               Airport_Name  ...    Purpose_of_flight Air_carrier  \
117  ARANSAS COUNTY AIRPORT  ...             Personal         NaN   
118  ARANSAS COUNTY AIRPORT  ...  Executive/corporate         NaN   
153                     NaN  ...             Personal         NaN   
158                     NaN  ...             Personal         NaN   
159                     NaN  ...             Personal         NaN   
160                     NaN  ...             Personal         NaN   
242  MEDFORD-JACKSON COUNTY  ...   Aerial Application         NaN   
244           RIED HILLVIEW  ...             Personal         NaN   
245  MEDFORD-JACKSON COUNTY  ...             Personal         NaN   
248           RIED HILLVIEW  ...             Personal         NaN   

    Total_Fatal_Injuries Total_Serious_Injuries Total_Minor_Injuries  \
117                  3.0                    0.0                  0.0   
118                  3.0                    0.0                  0.0   
153                  2.0                    0.0                  4.0   
158                  0.0                    0.0                  0.0   
159                  0.0                    0.0                  0.0   
160                  2.0                    0.0                  4.0   
242                  0.0                    0.0                  0.0   
244                  0.0                    0.0                  0.0   
245                  0.0                    0.0                  0.0   
248                  0.0                    0.0                  0.0   

    Total_Uninjured Weather_Condition  Broad_phase_of_flight   Report_Status  \
117             0.0               IMC               Approach  Probable Cause   
118             0.0               IMC               Approach  Probable Cause   
153             0.0               VMC                Unknown  Probable Cause   
158             3.0               VMC                 Cruise  Probable Cause   
159             3.0               VMC               Approach  Probable Cause   
160             0.0               VMC                 Cruise  Probable Cause   
242             3.0               VMC                   Taxi  Probable Cause   
244             3.0               VMC               Standing  Probable Cause   
245             3.0               VMC                   Taxi  Probable Cause   
248             3.0               VMC                   Taxi  Probable C

I see here that though these duplicate rows do represent separate aircraft in multi-aircraft incidents, the injury and/or fatality numbers are combined. This would constitute duplicate numbers in certain columns that would render errors in the analysis.

So let's remove the duplicates from this subset.

In [162]:
# remove the duplicate rows using the Event_Id column
df = df.drop_duplicates(subset=['Event_Id'], keep='first')

In [163]:
# check for duplicates again in Event_Id column
duplicate_rows_events = df[df.duplicated(subset=['Event_Id'], keep=False)]

duplicate_rows_events.info()

<class 'pandas.core.frame.DataFrame'>
Index: 0 entries
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event_Id                0 non-null      object 
 1   Investigation_Type      0 non-null      object 
 2   Accident_Number         0 non-null      object 
 3   Event_Date              0 non-null      object 
 4   Location                0 non-null      object 
 5   Country                 0 non-null      object 
 6   Latitude                0 non-null      object 
 7   Longitude               0 non-null      object 
 8   Airport_Code            0 non-null      object 
 9   Airport_Name            0 non-null      object 
 10  Injury_Severity         0 non-null      object 
 11  Aircraft_damage         0 non-null      object 
 12  Aircraft_Category       0 non-null      object 
 13  Registration_Number     0 non-null      object 
 14  Make                    0 non-null      object 
 15  Model 

In [164]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 87951 entries, 0 to 88888
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event_Id                87951 non-null  object 
 1   Investigation_Type      87951 non-null  object 
 2   Accident_Number         87951 non-null  object 
 3   Event_Date              87951 non-null  object 
 4   Location                87899 non-null  object 
 5   Country                 87729 non-null  object 
 6   Latitude                34212 non-null  object 
 7   Longitude               34203 non-null  object 
 8   Airport_Code            49484 non-null  object 
 9   Airport_Name            52031 non-null  object 
 10  Injury_Severity         86961 non-null  object 
 11  Aircraft_damage         84848 non-null  object 
 12  Aircraft_Category       32181 non-null  object 
 13  Registration_Number     86601 non-null  object 
 14  Make                    87888 non-null  obj

In [165]:
# check for duplicate rows in the Accident_Number column to verify there are no more duplicates
duplicate_rows_accidents = df[df.duplicated(subset=['Accident_Number'], keep=False)]

if duplicate_rows_accidents.empty:
  print("No duplicate rows found.")
else:
  print("Duplicate rows found.")

No duplicate rows found.


Remove certain columns that are mostly empty and would not contain data useful to the intended analysis.

I propose removing Latitude, Longitude, Schedule, and Air_carrier as those columns are mostly empty and would not contribute to the analysis.

In [166]:
# remove Latitude, Longitude, Schedule, and Air_carrier columns

df = df.drop(['Latitude', 'Longitude', 'Schedule', 'Air_carrier'], axis=1)

df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 87951 entries, 0 to 88888
Data columns (total 27 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event_Id                87951 non-null  object 
 1   Investigation_Type      87951 non-null  object 
 2   Accident_Number         87951 non-null  object 
 3   Event_Date              87951 non-null  object 
 4   Location                87899 non-null  object 
 5   Country                 87729 non-null  object 
 6   Airport_Code            49484 non-null  object 
 7   Airport_Name            52031 non-null  object 
 8   Injury_Severity         86961 non-null  object 
 9   Aircraft_damage         84848 non-null  object 
 10  Aircraft_Category       32181 non-null  object 
 11  Registration_Number     86601 non-null  object 
 12  Make                    87888 non-null  object 
 13  Model                   87859 non-null  object 
 14  Amateur_Built           87851 non-null  obj

## Aircraft_Category
The column for Aircraft_Category is also mostly empty, but that data could be useful. The business is after all seeking data related to airplanes specifically, so removing the column entirely would not work well and removing rows that are not airplanes and keeping those that are would be beneficial. Simply removing all rows that do not have a category entry would greatly reduce the number of total rows available for analysis, and most of those removed would likely be airplanes.

I would like to explore the idea of filling in as many of the missing values as I can. This could be done to some extent by making use of the Make column. For example, for all Aircraft_Category entries that are present, if the Make entry is Cessna, then the Aircraft_Category is invariably Airplane. So the missing category entries for Cessnas could reasonably be filled in. The same is true for other Makes like Sikorsky (Helicopter).

See below for an example:

In [167]:
# Show the Aircraft_Category values for Cessna in the Make column
df[df['Make'] == 'Cessna']['Aircraft_Category'].unique()

array([nan, 'Airplane'], dtype=object)

In [168]:
# Show how many nan Aircraft_Category values there are for Cessna
df[df['Make'] == 'Cessna']['Aircraft_Category'].isna().sum()

18344

In [169]:
# Show how many 'Airplane' Aircraft_Category values there are for Cessna
df[(df['Make'] == 'Cessna') & (df['Aircraft_Category'] == 'Airplane')]['Aircraft_Category'].count()

3581

So we can add another 18344 airplane entries to our data by filling in the missing value for Cessna in the Aircraft_Category column

In [170]:
# Show the Aircraft_Category values for Skiorsky in the Make column
df[df['Make'] == 'Sikorsky']['Aircraft_Category'].unique()

array(['Helicopter', nan], dtype=object)

In [171]:
# Show how many nan Aircraft_Category values there are for Sikorsky
df[df['Make'] == 'Sikorsky']['Aircraft_Category'].isna().sum()

128

And here we would be able to remove 128 additional rows for Sikorsky helicopters.

In [172]:
# Fill in Aircraft_Category as 'Airplane' for Cessna
df.loc[df['Make'] == 'Cessna', 'Aircraft_Category'] = 'Airplane'

In [173]:
# Show the Aircraft_Category values for Cessna in the Make column
df[df['Make'] == 'Cessna']['Aircraft_Category'].unique()

array(['Airplane'], dtype=object)

In [174]:
# Show how many 'Airplane' Aircraft_Category values there are for Cessna
df[(df['Make'] == 'Cessna') & (df['Aircraft_Category'] == 'Airplane')]['Aircraft_Category'].count()

21925

So now, instead of only 3500 Cessna airplanes, we have almost 22000 entries, greatly increasing the verified airplane subset.

So here I will continue finding Makes that are airplanes only, and filling in the missing values.

In [175]:
# print a list of the unique values in the Make column
df['Make'].value_counts()

Make
Cessna           21925
Piper            11903
CESSNA            4914
Beech             4290
PIPER             2841
                 ...  
Geertz               1
Conrad Menzel        1
Blucher              1
Gideon               1
ROYSE RALPH L        1
Name: count, Length: 8202, dtype: int64

I realize here that I need to do some further cleaning of the Make column so Cessna and CESSNA (and other similar issues) are not separate values.

## Clean up the Make column

In [176]:
#Fill Nan values in Make first
df['Make'] = df['Make'].fillna('')

# Clean the Make column for misspellings, all caps issues, etc
make_column_name_replace = {'Ab Sportine Aviacija': 'Sportine Aviacija', 'AB SPORTINE AVIACIJA': 'Sportine Aviacija', 'SPORTINE AVIACIJA': 'Sportine Aviacija', 'Abrams/nuding': 'Abrams', 'ACRO': 'Acro Sport', 'Adams': 'Adams Balloon', 'ADAMS': 'Adams Balloon', 'ADAMS BALLOONS LLC': 'Adams Balloon', 'AERO COMMANDER': 'Aero Commander', 'AERO VODOCHODY': 'Aero Vodochody', 'AEROVODOCHODY': 'Aero Vodochody', 'Aero Vodochody Aero Works': 'Aero Vodochody', 'AEROFAB INC': 'Aerofab Inc.', 'AEROMOT': 'Aeromot', 'AERONCA': 'Aeronca', 'Aeronca Aircraft Corporation': 'Aeronca', 'AEROPRO CZ': 'Aeropro CZ', 'AEROS': 'Aeros', 'Aeros LTD': 'Aeros', 'AEROS LTD': 'Aeros', 'AEROSPATIALE': 'Aerospatiale', 'AEROSTAR': 'Aerostar', 'Aerostar International': 'Aerostar', 'AEROSTAR INTERNATIONAL': 'Aerostar', 'Aerostar International Inc': 'Aerostar', 'AEROSTAR INTERNATIONAL INC': 'Aerostar', 'Aerostar International Inc.': 'Aerostar', 'Aerostar International, Inc.': 'Aerostar', 'AEROTEK': 'Aerotek', 'Aerotek-pitts': 'Aerotek', 'AEROTEK INC': 'Aerotek', 'AGUSTA': 'Agusta', 'AGUSTA AEROSPACE CORP': 'Agusta', 'AGUSTA SPA': 'Agusta', 'Agusta Spa': 'Agusta', 'Agusta-bell': 'Agusta', 'Agusta/Westland': 'Agusta', 'AGUSTAWESTLAND': 'Agusta', 'AgustaWestland': 'Agusta', 'AgustadWestland': 'Agusta', 'AGUSTAWESTLAND PHILADELPHIA': 'Agusta', 'AGUSTAWESTLAND SPA': 'Agusta', 'AIR CREATION': 'Air Creation', 'Air Creations': 'Air Creation', 'AIR TRACTOR': 'Air Tractor', 'AIR TRACTOR INC': 'Air Tractor', 'Air Tractor Inc': 'Air Tractor', 'Air Tractor Inc.': 'Air Tractor', 'AIR TRACTOR INC.': 'Air Tractor', 'Air Tractor, Inc.': 'Air Tractor', 'Air Borne Windsports Pty. Ltd.': 'Airborne', 'AIRBORNE (AUSTRALIA)': 'Airborne', 'AIRBORNE AUSTRALIA': 'Airborne', 'AIRBORNE EXTREME LLC': 'Airborne', 'AirBorne WindSport': 'Airborne', 'Airborne Windsports': 'Airborne', 'AIRBORNE WINDSPORTS': 'Airborne', 'Airborne Windsports Ltd': 'Airborne', 'AIRBORNE WINDSPORTS PTY LTD': 'Airborne', 'Airborne Windsports PTY LTD': 'Airborne', 'AIRBORNE': 'Airborne', 'AIRBUS': 'Airbus', 'AIRBUS HELICOPTERS': 'Airbus Helicopters', 'AIRBUS HELICOPTER': 'Airbus Helicopters', 'Airbus Helicopters (Eurocopte': 'Airbus Helicopters', 'Airbus Helicopters Deutschland': 'Airbus Helicopters', 'AIRBUS HELICOPTERS INC': 'Airbus Helicopters', 'Airbus Industrie': 'Airbus', 'AIRBUS INDUSTRIE': 'Airbus', 'AIRCRAFT MFG & DEVELOPMENT CO': 'Aircraft Mfg & Dev. Co.', 'Aircraft Mfg & Dev. Co. (amd)': 'Aircraft Mfg & Dev. Co.', 'Aircraft Mfg & Dev. Co. (AMD)':'Aircraft Mfg & Dev. Co.', 'Aircraft Mfg & Development Co.': 'Aircraft Mfg & Dev. Co.', 'AIRCRAFT MFG & DVLPMT CO': 'Aircraft Mfg & Dev. Co.', 'ALON': 'Alon', 'AMERICAN': 'American Aviation', 'American': 'American Aviation', 'AMERICAN AVIATION': 'American Aviation', 'American Aviation Corp. (aac)': 'American Aviation', 'AMERICAN CHAMPION': 'American Champion', 'American Champion (acac)': 'American Champion', 'American Champion (ACAC)': 'American Champion', 'American Champion Aircraft': 'American Champion', 'AMERICAN CHAMPION AIRCRAFT': 'American Champion', 'American Champion Aircraft Cor': 'American Champion', 'AMERICAN EUROCOPTER CORP': 'American Eurocopter', 'AMERICAN GENERAL ACFT CORP': 'American General Aircraft', 'American Legand Aircraft': 'American Legend', 'AMERICAN LEGEND': 'American Legend', 'AMERICAN LEGEND AIRCRAFT CO': 'American Legend', 'American Legend Aircraft Co.': 'American Legend', 'Anderson': 'Anderson Aircraft Corp.', 'Atr': 'ATR', 'AVIAT': 'Aviat', 'AVIAT AIRCRAFT': 'Aviat', 'Aviat Aircraft Inc': 'Aviat', 'AVIAT AIRCRAFT INC': 'Aviat', 'Aviat Aircraft Inc.': 'Aviat', 'Aviat Aircraft, Inc.': 'Aviat', 'Aviat Inc': 'Aviat', 'AVIAT INC': 'Aviat', 'Avid': 'Avid Aircraft', 'AVID': 'Avid Aircraft', 'AYERS': 'Ayres', 'Ayers': 'Ayres', 'AYRES': 'Ayres', 'AYRES CORP': 'Ayres', 'Ayres Corporation': 'Ayres', 'AYRES CORPORATION': 'Ayres', 'Bede': 'Bede Aircraft', 'BEDE': 'Bede Aircraft', 'BEECH': 'Beech', 'BEECH AIRCRAFT': 'Beech', 'BEECH AIRCRAFT CO.': 'Beech', 'Beech Aircraft Corp': 'Beech', 'Beech Aircraft Corporation': 'Beech', 'BEECH AIRCRAFT CORPORATION': 'Beech', 'Beechcraft': 'Beech', 'BEECHCRAFT': 'Beech', 'Beechcraft Corporation': 'Beech', 'BOEING': 'Boeing', 'Boeing - Canada (de Havilland)': 'Boeing', 'Boeing (Stearman)': 'Boeing', 'BOEING 777-306ER': 'Boeing', 'Boeing Commercial Airplane Gro': 'Boeing', 'BOEING COMPANY': 'Boeing', 'Boeing Company': 'Boeing', 'BOEING OF CANADA/DEHAV DIV': 'Boeing', 'Boeing Stearman': 'Boeing', 'BOEING-STEARMAN': 'Boeing', 'Boeing-brown': 'Boeing', 'BOMBARDIER': 'Bombardier', 'Bombardier Aerospace, Inc.': 'Bombardier', 'Bombardier Canadair': 'Bombardier', 'BOMBARDIER INC': 'Bombardier', 'BOMBARDIER LEARJET CORP.': 'Bombardier', 'Bombardier, Inc.': 'Bombardier', 'BRITISH AEROSPACE': 'British Aerospace', 'British Aerospace Civil Aircr': 'British Aerospace', 'BRITTEN NORMAN': 'Britten Norman', 'Britten-norman': 'Britten Norman', 'BRITTEN-NORMAN': 'Britten Norman', 'CANADAIR': 'Canadair', 'CANADAIR LTD': 'Canadair', 'CASA': 'Casa', 'Cassult Racer': 'Cassuit', 'CASSUTT': 'Cassuit', 'Cesna': 'Cessna', 'CESSNA': 'Cessna', 'CESSNA AIRCRAFT': 'Cessna', 'CESSNA Aircraft': 'Cessna', 'CESSNA AIRCRAFT CO': 'Cessna', 'Cessna Aircraft Co.': 'Cessna', 'CESSNA AIRCRAFT COMPANY': 'Cessna', 'Cessna Aircraft Company': 'Cessna', 'Cessna Ector': 'Cessna', 'CESSNA ECTOR': 'Cessna', 'Cessna Reems': 'Cessna', 'CESSNA REIMS': 'Cessna', 'Cessna Robertson': 'Cessna', 'Cessna Skyhawk II': 'Cessna', 'Cessna Soloy': 'Cessna', 'Cessna Wren': 'Cessna', 'CESSNA/AIR REPAIR INC': 'Cessna', 'CESSNA/WEAVER': 'Cessna', 'CHALLENGER': 'Challenger', 'Challenger Ii': 'Challenger', 'CHAMBERLAIN GERALD': 'Chamberlain', 'CHAMBERLIN VICTOR WAYNE': 'Chamberlain', 'CHAMPION': 'Champion', 'CHANCE VOUGHT': 'Chance Vought', 'CHRISTEN INDUSTRIES INC': 'Christen Industries', 'Christen Industries, Inc.': 'Christen Industries', 'CIRRUS': 'Cirrus', 'Cirrus Design': 'Cirrus', 'CIRRUS DESIGN': 'Cirrus', 'Cirrus Design Corp': 'Cirrus', 'CIRRUS DESIGN CORP': 'Cirrus', 'Cirrus Design Corp.': 'Cirrus', 'CIRRUS DESIGN CORP.': 'Cirrus', 'Cirrus Design Corporation': 'Cirrus', 'CIRRUS DESIGN CORPORATION': 'Cirrus', 'CLASSIC AIRCRAFT CORP': 'Classic Aircraft Corp', 'Classic Aircraft Corp.': 'Classic Aircraft Corp', 'COLUMBIA': 'Columbia', 'Columbia Aircraft': 'Columbia', 'Columbia Aircraft Mfg': 'Columbia', 'COLUMBIA AIRCRAFT MFG': 'Columbia', 'Columbia Aircraft Mfg.': 'Columbia', 'COMMANDER': 'Commander', 'Commander Aircraft': 'Commander', 'COMMANDER AIRCRAFT CO': 'Commander', 'Commander Aircraft Company': 'Commander', 'CONSOLIDATED AERONAUTICS': 'Consolidated Aero', 'Consolidated Aeronautics Inc.': 'Consolidated Aero', 'CONSOLIDATED AERONAUTICS INC.': 'Consolidated Aero', 'Consolidated Aeronautics, Inc': 'Consolidated Aero', 'Consolidated Aeronautics, Inc.': 'Consolidated Aero', 'CONSOLIDATED VULTEE': 'Consolidated Aero', 'Consolidated-vultee': 'Consolidated Aero', 'CONVAIR': 'Convair', 'Convair Div. Of Gen. Dynamics': 'Convair', 'COSTRUZIONI AERONAUTICHE TECNA': 'Costruzioni', 'Costruzioni Aeronautiche': 'Costruzioni', 'Costruzioni AeronauticheTecnam': 'Costruzioni', 'CUB CRAFTER': 'Cub Crafters', 'CUB CRAFTERS': 'Cub Crafters', 'CUB CRAFTERS INC': 'Cub Crafters', 'Cub Crafters Inc': 'Cub Crafters', 'Cub Crafters Inc.': 'Cub Crafters', 'Cub Crafters, Inc.': 'Cub Crafters', 'Cubcrafter': 'Cub Crafters', 'Cubcrafters': 'Cub Crafters', 'CUBCRAFTERS': 'Cub Crafters', 'CubCrafters': 'Cub Crafters', 'CUBCRAFTERS INC': 'Cub Crafters', 'CubCrafters Inc': 'Cub Crafters', 'Cubcrafters, Inc': 'Cub Crafters', 'CubCrafters, Inc': 'Cub Crafters', 'CULVER': 'Culver', 'CULVER GLENN': 'Culver', 'Culver-revolution': 'Culver', 'Culver, Aurther L.': 'Culver', 'CURTIS JOHN P': 'Curtiss Wright', 'Curtis-travel Air': 'Curtiss Wright', 'Curtis-wright': 'Curtiss Wright', 'CURTISS': 'Curtiss Wright', 'CURTISS WRIGHT': 'Curtiss Wright', 'Curtiss-wright': 'Curtiss Wright', 'Curtiss-Wright': 'Curtiss Wright', 'CZECH': 'Czech Aircraft Works', 'CZECH AIRCRAFT WORKS': 'Czech Aircraft Works', 'CZECH AIRCRAFT WORKS SPOL SRO': 'Czech Aircraft Works', 'Czech Aircraft Works SPOL SRO': 'Czech Aircraft Works', 'Czech Aircraft Works Spol Sro': 'Czech Aircraft Works', 'Czech Sport Aircraft': 'Czech Sport', 'CZECH SPORT AIRCRAFT A S': 'Czech Sport', 'Czech Sport Aircraft a.s.': 'Czech Sport', 'Czech Sport Aircraft AS': 'Czech Sport', 'CZECH SPORT AIRCRAFT AS': 'Czech Sport', 'CZECH SPORTPLANES SRO': 'Czech Sport', 'DASSAULT': 'Dassault', 'Dassault Aviation': 'Dassault', 'DASSAULT AVIATION': 'Dassault', 'Dassault Falcon': 'Dassault', 'Dassault-breguet': 'Dassault', 'DASSAULT-BREGUET': 'Dassault', 'Dassault-Breguet': 'Dassault', 'Dassault/sud': 'Dassault', 'DASSAULT/SUD': 'Dassault', 'DE HAVILLAND': 'de Havilland', 'De Havilland': 'de Havilland', 'Dehavilland': 'de Havilland', 'DEHAVILLAND': 'de Havilland', 'DeHavilland': 'de Havilland', 'DEHAVILLAND CANADA': 'de Havilland', 'DIAMOND': 'Diamond', 'Diamond Aicraft Industries Inc': 'Diamond', 'Diamond Aircraft': 'Diamond', 'DIAMOND AIRCRAFT': 'Diamond', 'DIAMOND AIRCRAFT IND GMBH': 'Diamond', 'Diamond Aircraft Industries': 'Diamond', 'DIAMOND AIRCRAFT INDUSTRIES': 'Diamond', 'DIAMOND AIRCRAFT INDUSTRIES IN': 'Diamond', 'Diamond Aircraft Industry Inc': 'Diamond', 'DORNIER': 'Dornier', 'DORNIER GMBH': 'Dornier', 'DOUGLAS': 'Douglas', 'DOWNER': 'Downer', 'Downer Aircraft Industries': 'Downer', 'EAGLE': 'Eagle', 'Eagle (ultralight)': 'Eagle', 'EAGLE AIRCRAFT CO': 'Eagle', 'Eagle Aircraft Co.': 'Eagle', 'ECLIPSE AVIATION': 'Eclipse Aviation', 'ECLIPSE AVIATION CORP': 'Eclipse Aviation', 'Eclipse Aviation Corporation': 'Eclipse Aviation', 'EDGE': 'Edge', 'EIPPER': 'Eipper', 'Eipper Formance': 'Eipper', 'EIPPER FORMANCE INC': 'Eipper', 'Eipper Mx Ii Quicksilver': 'Eipper', 'Eipper Quicksilver': 'Eipper', 'Eipper Quicksiver E': 'Eipper', 'Eipper-formance': 'Eipper', 'EMBRAER': 'Embraer', 'Embraer Aircraft': 'Embraer', 'EMBRAER EXECUTIVE AIRCRAFT INC': 'Embraer', 'EMBRAER S A': 'Embraer', 'EMBRAER-EMPRESA BRASILEIRA DE': 'Embraer', 'ENGINEERING & RESEARCH': 'Engineering & Research', 'Engineering and Research': 'Engineering & Research', 'ENSTROM': 'Enstrom', 'ENSTROM HELICOPTER CORP': 'Enstrom', 'ERCOUPE': 'Ercoupe', 'Ercoupe (eng & Research Corp.)': 'Ercoupe', 'EUROCOPTER': 'Eurocopter', 'Eurocopter Deutsch': 'Eurocopter', 'Eurocopter Deutschland': 'Eurocopter', 'Eurocopter Deutschland Gmbh': 'Eurocopter', 'EUROCOPTER DEUTSCHLAND GMBH': 'Eurocopter', 'Eurocopter France': 'Eurocopter', 'EUROCOPTER FRANCE': 'Eurocopter', 'EUROPA': 'Europa', 'Europa Aviation Inc': 'Europa', 'EVEKTOR AEROTECHNIK': 'Evektor Aerotechnik', 'Evektor Aerotechnik': 'Evektor Aerotechnik', 'Evektor Aerotechnik AS': 'Evektor Aerotechnik', 'Evektor-aerotechnik': 'Evektor Aerotechnik', 'Evektor-aerotechnik A.s.': 'Evektor Aerotechnik', 'Evektor-aerotechnik As': 'Evektor Aerotechnik', 'EVEKTOR-AEROTECHNIK AS': 'Evektor Aerotechnik', 'Evektor-Aerotechnik AS': 'Evektor Aerotechnik', 'EXTRA': 'Extra', 'Extra Flugzeugbau': 'Extra', 'EXTRA FLUGZEUGBAU': 'Extra', 'Extra Flugzeugbau Gmbh': 'Extra', 'EXTRA FLUGZEUGBAU GMBH': 'Extra', 'EXTRA Flugzeugproduktions-GMBH': 'Extra', 'Extra Flugzeugproduktions-und': 'Extra', 'EXTRA FLUGZEUGPRODUKTIONS-UND': 'Extra', 'Extra Flugzeugrau Gmbh': 'Extra', 'FAIRCHILD': 'Fairchild', 'Fairchild Industries': 'Fairchild', 'Fairchild Swearingen': 'Fairchild', 'Fairchild/swearingen': 'Fairchild', 'FAIRCHILD(HOWARD)': 'Fairchild', 'Fairchild Heli-porter': 'Fairchild', 'FAIRCHILD HELI-PORTER': 'Fairchild', 'Fairchild-heliporter': 'Fairchild', 'FAIRCHILD HILLER': 'Fairchild Hiller', 'FANTASY AIR': 'Fantasy Air', 'FANTASY AIR SRO': 'Fantasy Air', 'Fantasy Sky Promotions': 'Fantasy Air', 'FIELDS': 'Fields', 'FISHER': 'Fisher', 'Fischer': 'Fisher', 'Fisher Aero': 'Fisher', 'Fisher Flying Products': 'Fisher', 'FISHER FLYING PRODUCTS': 'Fisher', 'FISHER HAROLD R': 'Fisher', 'FISHER MICHAEL E': 'Fisher', 'FISHER MICHAEL H': 'Fisher', 'FLEET': 'Fleet', 'FLIGHT DESIGN': 'Flight Design', 'FLIGHT DESIGN GENERAL AVN GMBH': 'Flight Design', 'Flight Design Gmbh': 'Flight Design', 'FLIGHT DESIGN GMBH': 'Flight Design', 'Flight Design GMBH': 'Flight Design', 'Flight Star': 'Flightstar', 'FLIGHTSTAR': 'Flightstar', 'FLIGHTstar': 'Flightstar', 'FLIGHTSTAR INC': 'Flightstar', 'FLIGHTSTAR SPORTPLANES': 'Flightstar', 'Flightstar Sportplanes': 'Flightstar', 'FOKKER': 'Fokker', 'FOUND ACFT CANADA INC': 'Found Aircraft Canada', 'Found Acft': 'Found Aircraft Canada', 'Found Aircraft Canada Inc': 'Found Aircraft Canada', 'GARLICK': 'Garlick', 'Garlick Helicipters Inc.': 'Garlick', 'GARLICK HELICOPTERS INC': 'Garlick', 'Garlick Helicopters Inc.': 'Garlick', 'Gates Lear Jet': 'Gates Learjet', 'GATES LEAR JET': 'Gates Learjet', 'GATES LEAR JET CORP.': 'Gates Learjet', 'GATES LEARJET': 'Gates Learjet', 'GATES LEARJET CORP': 'Gates Learjet', 'GATES LEARJET CORP.': 'Gates Learjet', 'Gates Learjet Corporation':'Gates Learjet', 'GENERAL ATOMICS': 'General Atomics', 'GLASAIR': 'Glasair', 'Glasair Aviation LLC': 'Glasair', 'GLASAIR AVIATION USA LLC': 'Glasair', 'Glasair Iii': 'Glasair', 'Glassair': 'Glasair', 'Glassair Iii': 'Glasair', 'GLOBE': 'Globe', 'Globe Swift': 'Globe', 'GREAT LAKES': 'Great Lakes', 'Great Lakes Adams': 'Great Lakes', 'Great Lakes Aircraft Company': 'Great Lakes', 'GRUMMAN': 'Grumman', 'Grumman Acft Eng': 'Grumman', 'GRUMMAN ACFT ENG COR': 'Grumman', 'GRUMMAN ACFT ENG COR-SCHWEIZER': 'Grumman', 'GRUMMAN AIRCRAFT': 'Grumman', 'GRUMMAN AIRCRAFT COR-SCHWEIZER': 'Grumman', 'GRUMMAN AIRCRAFT ENG CORP': 'Grumman', 'Grumman American': 'Grumman', 'GRUMMAN AMERICAN': 'Grumman', 'Grumman American Aviation': 'Grumman', 'GRUMMAN AMERICAN AVIATION CORP': 'Grumman', 'GRUMMAN AMERICAN AVN. CORP': 'Grumman', 'Grumman American Avn. Corp.': 'Grumman', 'GRUMMAN AMERICAN AVN. CORP.': 'Grumman', 'GRUMMAN American Corporation': 'Grumman', 'Grumman Schweizer': 'Grumman', 'GRUMMAN SCHWEIZER': 'Grumman', 'Grumman-schweizer': 'Grumman', 'Grumman-Schweizer': 'Grumman', 'GULFSTREAM': 'Gulfstream', 'Gulfstream Aerospace': 'Gulfstream', 'GULFSTREAM AEROSPACE': 'Gulfstream', 'Gulfstream Aerospace Corp': 'Gulfstream', 'Gulfstream Aerospace Corp.': 'Gulfstream', 'Gulfstream Aerospace LP': 'Gulfstream', 'GULFSTREAM AM CORP COMM DIV': 'Gulfstream', 'Gulfstream American': 'Gulfstream', 'Gulfstream American Corp': 'Gulfstream', 'GULFSTREAM AMERICAN CORP': 'Gulfstream', 'Gulfstream American Corp.': 'Gulfstream', 'GULFSTREAM AMERICAN CORP.': 'Gulfstream', 'GULFSTREAM SCHWEIZER A/C CORP': 'Gulfstream', 'Gulfstream-schweizer': 'Gulfstream', 'GULFSTREAM-SCHWEIZER': 'Gulfstream', 'HALL': 'Hall', 'Hall Christen Eagle': 'Hall', 'HALL DON H': 'Hall', 'HALL JEFFREY': 'Hall', 'HALL STEVEN C': 'Hall', 'HALL TERRENCE / HALL CATHIE': 'Hall', 'HALL THOMAS K': 'Hall', 'Hall-cavalier': 'Hall', 'HALL, WENDALL W': 'Hall', 'Hamilton Stoddard': 'Hamilton', 'HAWKER': 'Beech', 'HAWKER AIRCRAFT LTD': 'Beech', 'Hawker Aircraft Ltd': 'Beech', 'Hawker Aircraft Ltd.': 'Beech', 'Hawker Beech': 'Beech', 'HAWKER BEECH': 'Beech', 'Hawker Beechcraft': 'Beech', 'HAWKER BEECHCRAFT': 'Beech', 'HAWKER BEECHCRAFT CORP': 'Beech', 'Hawker Beechcraft Corp.': 'Beech', 'Hawker Beechcraft Corporation': 'Beech', 'HAWKER BEECHCRAFT CORPORATION': 'Beech', 'Hawker Siddeley': 'Hawker', 'HAWKER SIDDELEY': 'Hawker', 'Hawker Siddely': 'Hawker', 'Hawker-beechcraft': 'Beech', 'Hawker-Beechcraft': 'Beech', 'Hawker-Beechcraft Corporation': 'Beech', 'Helie': 'Helio', 'HELIO': 'Helio', 'Helio Aircraft Ltd': 'Helio', 'HILLER': 'Hiller', 'Hiller-osborn': 'Hiller', 'HILLER-ROGERSON HELICOPTER': 'Hiller', 'Hiller-soloy': 'Hiller', 'HILLER-TRI-PLEX IND.INC.': 'Hiller', 'HONDA AIRCRAFT': 'Honda Aircraft', 'HONDA AIRCRAFT CO LLC': 'Honda Aircraft', 'Honda Jet': 'Honda Aircraft', 'HOWARD AIRCRAFT': 'Howard Aircraft', 'Howard Aircraft Corp.': 'Howard Aircraft', 'HUGHES': 'Hughes', 'HUGHES HELICOPTERS INC': 'Hughes', 'HUGHES/HELICOPTER ASSOCS INC': 'Hughes', 'HUNTER': 'Hunter', 'HUNTER GEORGE': 'Hunter', 'ICON': 'Icon', 'ICON AIRCRAFT INC': 'Icon', 'INTERPLANE': 'Interplane', 'Interplane Llc': 'Interplane', 'INTERPLANE S R O': 'Interplane', 'Interplane Sro': 'Interplane', 'INTERSTATE': 'Interstate', 'ISRAEL AEROSPACE INDUSTRIESLTD': 'Israel Aircraft Industries', 'ISRAEL AIRCRAFT INDUSTRIES': 'Israel Aircraft Industries', 'JABIRU': 'Jabiru Aircraft', 'JABIRU AIRCRAFT PTY LTD': 'Jabiru Aircraft', 'JABIRU USA SPORT AIRCRAFT': 'Jabiru Aircraft', 'JABIRU USA SPORT AIRCRAFT LLC': 'Jabiru Aircraft', 'JABIRU USA SPORT AIRCRAFT, LLC': 'Jabiru Aircraft', 'JACKSON': 'Jackson', 'JACKSON DENNIS': 'Jackson', 'JACKSON FRED M': 'Jackson', 'JACKSON JEREMIAH D': 'Jackson', 'JONES': 'Jones', 'JONES KENT C': 'Jones', 'JONES PETER M': 'Jones', 'JONES RALPH D': 'Jones', 'JONES RODNEY V': 'Jones', 'JONES RONALD C': 'Jones', 'JUST': 'Just', 'JUST AIRCRAFT': 'Just', 'Just Aircraft Llc': 'Just', 'JUST AIRCRAFT LLC': 'Just', 'Just Aircraft LLC': 'Just', 'KAMAN': 'Kaman', 'KAMAN AEROSPACE CORP': 'Kaman', 'KITFOX': 'Kitfox', 'Kitfox IV': 'Kitfox', 'Kitfox Ten, Inc': 'Kitfox', 'KOLB': 'Kolb', 'KOLB AIRCRAFT CO': 'Kolb', 'KOLB AIRCRAFT INC': 'Kolb', 'Kolb Company': 'Kolb', 'Kolb Twin Star': 'Kolb', 'LAKE': 'Lake', 'Lake John K': 'Lake', 'LANCAIR': 'Lancair', 'Lancair Company': 'Lancair', 'LANCAIR COMPANY': 'Lancair', 'Lanceair': 'Lancair', 'LARSEN': 'Larsen', 'Larsen Charles Bennett': 'Larsen', 'Larson': 'Larsen', 'LARSON': 'Larsen', 'LARSON KEN W': 'Larsen', 'LARSON ROGER H': 'Larsen', 'Larson Smith Miniplane': 'Larsen', 'Larson, C.h.': 'Larsen', 'LEARJET': 'Learjet', 'LEARJET INC': 'Learjet', 'Learjet Inc': 'Learjet', 'LET': 'Let', 'LIBERTY AEROSPACE INCORPORATED': 'Liberty', 'LIBERTY AEROSPACE': 'Liberty', 'Liberty Aerospace': 'Liberty', 'Liberty Aerospace Inc.': 'Liberty', 'Liberty Aerospace Incorporate': 'Liberty', 'Liberty Aerospace, Inc.': 'Liberty', 'LINDSTRAND': 'Lindstrand', 'Lindstrand Balloons': 'Lindstrand', 'LINDSTRAND BALLOONS': 'Lindstrand', 'LINDSTRAND BALLOONS USA': 'Lindstrand', 'LINSTRAND': 'Lindstrand', 'LOCKHEED': 'Lockheed', 'Lockheed-martin': 'Lockheed', 'LOCKWOOD': 'Lockwood', 'LOCKWOOD AIRCRAFT CORP': 'Lockwood', 'LONG': 'Long', 'Long E-z-e': 'Long', 'Long Pitts': 'Long', 'Long-ez': 'Long', 'Longjev': 'Long', 'LUSCOMBE': 'Luscombe', 'Luscombe Silvaire Aircraft Co.': 'Luscombe', 'MARTIN CHARLES A': 'Martin', 'Martin Company': 'Martin', 'MARTIN CURTIS': 'Martin', 'MARTIN EDWARD H': 'Martin', 'Martin-pitts': 'Martin', 'Martin/harris': 'Martin', 'MAULE': 'Maule', 'Maule Air Inc.': 'Maule', 'MAULE AIRCRAFT CORP': 'Maule', 'MAXAIR': 'Maxair', 'Maxair Aircraft Corp.': 'Maxair', 'MAXAIR DRIFTER': 'Maxair', 'MCCLISH': 'McClish', 'Mcclish': 'McClish', 'Mcdonald': 'Mcdonnell Douglas', 'Mcdonald Douglas': 'Mcdonnell Douglas', 'MCDONNELL DOUGLAS': 'Mcdonnell Douglas', 'MCDONNELL DOUGLAS AIRCRAFT CO': 'Mcdonnell Douglas', 'MCDONNELL DOUGLAS CORPORATION': 'Mcdonnell Douglas', 'Mcdonnell-douglas': 'Mcdonnell Douglas', 'MCDONNELL DOUGLAS HELI CO': 'Mcdonnell Douglas Helicopter', 'MCDONNELL DOUGLAS HELICOPTER': 'Mcdonnell Douglas Helicopter', 'McDonnell Douglas Helicopter C': 'Mcdonnell Douglas Helicopter', 'Mcdonnell Douglas Helicopters': 'Mcdonnell Douglas Helicopter', 'McDonnell Douglas Helicopters': 'Mcdonnell Douglas Helicopter', 'MD HELICOPTER': 'Md Helicopter', 'MD Helicopter Inc': 'Md Helicopter', 'MD Helicopters Inc': 'Md Helicopter', 'MD HELICOPTER INC': 'Md Helicopter', 'Md Helicopter Inc.': 'Md Helicopter', 'Md Helicopters': 'Md Helicopter', 'MD HELICOPTERS': 'Md Helicopter', 'MD HELICOPTERS INC': 'Md Helicopter', 'Md Helicopters Inc.': 'Md Helicopter', 'Md Helicopters, Inc.': 'Md Helicopter', 'MD HELICOPTERS, INC.': 'Md Helicopter', 'Messerschmitt-boelkow-blohm': 'Messerschmitt', 'MESSERSCHMITT-BOELKOW-BLOHM': 'Messerschmitt', 'Messerschmitt-bolkow-blohm': 'Messerschmitt', 'MESSERSCHMITT-BOLKOW-BLOHM': 'Messerschmitt', 'MEYER': 'Meyer', 'MEYERS': 'Meyers', 'Meyers Aircraft Co.': 'Meyers', 'MEYERS INDUSTRIES INC': 'Meyers', 'MILLER': 'Miller', 'Miller Air Sports': 'Miller', 'Mitchell Aircraft Corp': 'Mitchell', 'MITSUBISHI': 'Mitsubishi', 'MONOCOUPE': 'Monocoupe', 'Monocoupe Aircraft': 'Monocoupe', 'MOONEY': 'Mooney', 'Mooney Aircraft': 'Mooney', 'Mooney Aircraft Corp': 'Mooney', 'MOONEY AIRCRAFT CORP.': 'Mooney', 'Mooney Aircraft Corporation': 'Mooney', 'MOONEY AIRPLANE CO INC': 'Mooney', 'MOONEY AIRPLANE COMPANY, INC.': 'Mooney', 'MOONEY INTERNATIONAL CORP': 'Mooney', 'MOORE': 'Moore', 'MORAVAN': 'Moravan', 'MORRISEY': 'Morrisey', 'MORRISON': 'Morrison', 'Murphy Aircraft': 'Murphy', 'Murphy Aircraft Mfg, Ltd.': 'Murphy', 'Murphy-charles': 'Murphy', 'Murphey': 'Murphy', 'MYERS': 'Myers', 'NANCHANG': 'Nanchang', 'Nanchang China': 'Nanchang', 'NANCHANG CHINA': 'Nanchang', 'NAVAL AIRCRAFT FACTORY': 'Naval Aircraft Factory', 'Navy': 'Naval Aircraft Factory', 'NAVION': 'Navion', 'NELSON': 'Nelson', 'Nelson/nelson': 'Nelson', 'New Glasair': 'Glasair', 'NEW KOLB AIRCRAFT CO': 'Kolb', 'NEW KOLB AIRCRAFT CO LLC': 'Kolb', 'New Piper': 'Piper', 'NEW PIPER': 'Piper', 'NEW PIPER AIRCRAFT INC': 'Piper', 'New Piper Aircraft, Inc.': 'Piper', 'NEW STANDARD': 'New Standard', 'NORD': 'Nord', 'Nord (sncan)': 'Nord', 'Nord Aviation': 'Nord', 'NORTH AMERICAN': 'North American', 'North American Aviation Div.': 'North American', 'North American Rockwell': 'North American', 'North American Rockwell Corp.': 'North American', 'NORTH AMERICAN-MEDORE': 'North American', 'North American/aero Classics': 'North American', 'North American-aero Classics': 'North American', 'North American-barene': 'North American', 'North American-kenney': 'North American', 'North American-maslon': 'North American', 'NORTH AMERICAN/AERO CLASSICS': 'North American', 'NORTH AMERICAN/SCHWAMM': 'North American', 'NORTH AMERICAN/VICTORIA MNT LT': 'North American', 'NORTH WING': 'North Wing', 'NORTH WING DESIGN': 'North Wing', 'NORTH WING UUM INC': 'North Wing', 'North Wing UUM INC': 'North Wing', 'NORTH WING UUM INC.': 'North Wing', 'NORTHROP': 'Northrop', 'NORTHWING': 'Northwing', 'Northwing Design': 'Northwing', 'NORTHWING DESIGN': 'Northwing', 'PACIFIC AEROSPACE CORP LTD': 'Pacific Aerospace', 'PIAGGIO': 'Piaggio', 'Piaggio Aero Industries S.p.a.': 'Piaggio', 'PIAGGIO AERO INDUSTRIES SPA': 'Piaggio', 'Piaggio Industrie': 'Piaggio', 'PIETENPOL': 'Pietenpol', 'Pietenpol-grega': 'Pietenpol', 'PILATUS': 'Pilatus', 'Pilatus Aircraft': 'Pilatus', 'PILATUS AIRCRAFT LTD': 'Pilatus', 'Pilatus Britten-norman': 'Pilatus', 'PILATUS BRITTEN-NORMAN': 'Pilatus', 'PIPER': 'Piper', 'PIPER / LAUDEMAN': 'Piper', 'Piper Aerostar': 'Piper', 'Piper Aircraft': 'Piper', 'PIPER AIRCRAFT': 'Piper', 'PIPER AIRCRAFT CORPORATION': 'Piper', 'Piper Aircraft Corporation': 'Piper', 'PIPER AIRCRAFT INC': 'Piper', 'PIPER AIRCRAFT, INC.': 'Piper', 'Piper Aircraft, Inc.': 'Piper', 'Piper Cub Crafters': 'Piper', 'Piper Pawnee': 'Piper', 'Piper-aerostar': 'Piper', 'PIPER-HARRIS': 'Piper', 'Piper/cub Crafters': 'Piper', 'PIPER/CUB CRAFTERS': 'Piper', 'Piper/Cub Crafters': 'Piper', 'Piper/stevens': 'Piper', 'PIPER/WALLY\'S FLYERS INC': 'Piper', 'PIPISTREL': 'Pipistrel', 'PIPISTREL D O O': 'Pipistrel', 'PIPISTREL DOO AJDOVSCINA': 'Pipistrel', 'PIPISTREL ITALIA S R L': 'Pipistrel', 'Pipistrel Italia SRL': 'Pipistrel', 'PITCAIRN': 'Pitcairn', 'PITTS': 'Pitts', 'PITTS AEROBATICS': 'Pitts', 'Pitts Special': 'Pitts', 'PITTS SPECIAL': 'Pitts', 'Pitts Spl.': 'Pitts', 'PROGRESSIVE AERODYNE INC': 'Progressive Aerodyne', 'Progressive Aerodyne Inc.': 'Progressive Aerodyne', 'PZL BIELSKO': 'PZL', 'PZL MIELEC': 'PZL', 'PZL OKECIE': 'PZL', 'Pzl Okecie': 'PZL', 'Pzl Swidnik': 'PZL', 'PZL SWIDNIK': 'PZL', 'PZL Warszawa-Okecie': 'PZL', 'Pzl Warzawa-cnpsl': 'PZL', 'Pzl Warzawa-okecie': 'PZL', 'Pzl-bielsko': 'PZL', 'PZL-BIELSKO': 'PZL', 'Pzl-mielec': 'PZL', 'Pzl-okecie': 'PZL', 'Pzl-swidnik': 'PZL', 'PZL-Swidnik': 'PZL', 'PZL-SWIDNIK': 'PZL', 'QUAD CITY': 'Quad City', 'Quad City Aircraft': 'Quad City', 'Quad City Aircraft Corp': 'Quad City', 'Quad City Ultralight': 'Quad City', 'QUAD CITY ULTRALIGHT ACFT CORP': 'Quad City', 'Quad City Ultralight Aircraft': 'Quad City', 'QUAD CITY ULTRALIGHT CORP': 'Quad City', 'Quad City Ultralight, Llc': 'Quad City', 'QUAD CITY ULTRALIGHTS': 'Quad City', 'Quadcity': 'Quad City', 'QUEST': 'Quest', 'QUEST Aircraft Company': 'Quest', 'QUEST AIRCRAFT COMPANY LLC': 'Quest', 'QUESTAIR INC': 'Questair', 'Questair, Inc.': 'Questair', 'Questaire': 'Questair', 'QUICKIE': 'Quickie', 'Quickie Aircraft': 'Quickie', 'Quickie-myers': 'Quickie', 'Quick Silver': 'Quicksilver', 'QUICKSILVER': 'Quicksilver', 'Quicksilver Aircraft': 'Quicksilver', 'QUICKSILVER AIRCRAFT': 'Quicksilver', 'QUICKSILVER AIRCRAFT CO': 'Quicksilver', 'Quicksilver Aircraft Northeast': 'Quicksilver', 'QUICKSILVER EIPPER ACFT INC': 'Quicksilver', 'QUICKSILVER ENTERPRISES INC': 'Quicksilver', 'Quicksilver II': 'Quicksilver', 'Quicksilver Manufacturing': 'Quicksilver', 'QUICKSILVER MANUFACTURING INC': 'Quicksilver', 'QUICKSILVER MFG': 'Quicksilver', 'Quiksilver': 'Quicksilver', 'RANS': 'Rans', 'RANS AIRCRAFT': 'Rans', 'Rans Company': 'Rans', 'RANS DESIGNS INC': 'Rans', 'Rans Employee Flying Club': 'Rans', 'RANS EMPLOYEE FLYING CLUB': 'Rans', 'RANS INC': 'Rans', 'Rans Inc.': 'Rans', 'RANS S-12': 'Rans', 'Rans S-12 Airaile': 'Rans', 'Rans, Inc.': 'Rans', 'Rans/hine': 'Rans', 'Raven Industries': 'Raven', 'RAVEN INDUSTRIES INC': 'Raven', 'RAYTHEON': 'Raytheon', 'Raytheon Aircraft Company': 'Raytheon', 'RAYTHEON AIRCRAFT COMPANY': 'Raytheon', 'Raytheon Co': 'Raytheon', 'RAYTHEON COMPANY': 'Raytheon', 'Raytheon Corporate Jets': 'Raytheon', 'RAYTHEON CORPORATE JETS INC': 'Raytheon', 'REIMS': 'Reims', 'REims': 'Reims', 'Reims Aviation': 'Reims', 'Reims Aviation Cessna': 'Reims', 'REIMS AVIATION S.A.': 'Reims', 'REIMS AVIATION SA': 'Reims', 'Reims-Cessna': 'Reims', 'REIMS-CESSNA': 'Reims', 'REMOS ACFT GMBH FLUGZEUGBAU': 'Remos', 'Remos Aircraft GmbH': 'Remos', 'REMOS AIRCRAFT GMBH': 'Remos', 'Remos Aircraft GMBH': 'Remos', 'Remos Aircraft GMBH Flugzeugba': 'Remos', 'REMOS AIRCRAFT GMBH FLUGZEUGBA': 'Remos', 'REPUBLIC': 'Republic', 'ROBIN': 'Robin', 'ROBINSON': 'Robinson', 'Robinson Helicopter': 'Robinson', 'ROBINSON HELICOPTER': 'Robinson', 'ROBINSON HELICOPTER CO': 'Robinson', 'ROBINSON HELICOPTER CO INC': 'Robinson', 'Robinson Helicopter Co.': 'Robinson', 'Robinson Helicopter Company': 'Robinson', 'ROBINSON HELICOPTER COMPANY': 'Robinson', 'Robinson Helicopters': 'Robinson', 'ROCKWELL': 'Rockwell', 'Rockwell Comdr': 'Rockwell', 'Rockwell Commander': 'Rockwell', 'ROCKWELL COMMANDER': 'Rockwell', 'Rockwell Int\'t': 'Rockwell', 'Rockwell International': 'Rockwell', 'ROCKWELL INTERNATIONAL': 'Rockwell', 'Rockwell Intl': 'Rockwell', 'ROGERS': 'Rogers', 'ROLLADEN-SCHNEIDER': 'Rolladen Schneider', 'ROLLADEN SCHNEIDER OHG': 'Rolladen Schneider', 'Rolladen-schneider': 'Rolladen Schneider', 'Rolladen-schneider Gmbh': 'Rolladen Schneider', 'ROLLADEN-SCHNEIDER GMBH': 'Rolladen Schneider', 'ROLLADEN-SCHNEIDER OHG': 'Rolladen Schneider', 'Roof': 'Root', 'Root, Arthur T.': 'Root', 'ROSE': 'Rose', 'ROSE HERBERT D': 'Rose', 'Rose Rhinehart': 'Rose', 'Rose Wesley': 'Rose', 'Rose-Rhinehart': 'Rose', 'ROSS ALFRED K/ONEILL TERRENCE': 'Ross', 'ROSS H/HERRIOTT M': 'Ross', 'ROSS JONATHAN': 'Ross', 'Ross/stonecipher': 'Ross', 'ROTORWAY': 'Rotorway', 'Rotorway Aircraft, Inc.': 'Rotorway', 'Rotorway Executive': 'Rotorway', 'Rotoway International': 'Rotorway', 'RUTAN': 'Rutan', 'Rutan Aircraft Factory': 'Rutan', 'RYAN': 'Ryan', 'Ryan Aeronautical': 'Ryan', 'RYAN AERONAUTICAL': 'Ryan', 'Ryan Aeronautics': 'Ryan', 'RYAN JOHN STEFFEY': 'Ryan', 'RYAN W Gross': 'Ryan', 'Ryan-navion': 'Ryan', 'SAAB': 'Saab', 'Saab-fairchild': 'Saab', 'Saab-scania': 'Saab', 'SAAB-SCANIA': 'Saab', 'SAAB-SCANIA AB': 'Saab', 'Saab-scania Ab (saab)': 'Saab', 'Saab-Scania AB (Saab)': 'Saab', 'SCHWEIZER': 'Schweizer', 'Schweizer 300CBi': 'Schweizer', 'Schweizer Aircraft Corp': 'Schweizer', 'SCHWEIZER AIRCRAFT CORP': 'Schweizer', 'Schweizer Aircraft Corp.': 'Schweizer', 'Schweizer Sgs': 'Schweizer', 'Schweizer, N36289': 'Schweizer', 'SCHWEIZER(HUGHES)': 'Schweizer', 'SCHWEIZER(HUGHES)AIRCRAFT CORP': 'Schweizer', 'Scottish': 'Scottish Aviation', 'SCOTTISH AVIATION': 'Scottish Aviation', 'SHORT': 'Short', 'SHORT BROS': 'Short', 'Short Bros.': 'Short', 'SHORT BROS. & HARLAND': 'Short', 'Short Brothers': 'Short', 'SHORT BROTHERS & HARLAND LTD.': 'Short', 'SHORT BROTHERS PLC': 'Short', 'SIAI MARCHETTI': 'Siai Marchetti', 'Siai-marchetti': 'Siai Marchetti', 'SIAI-MARCHETTI': 'Siai Marchetti', 'Siai-Marchetti': 'Siai Marchetti', 'SIKORSKY': 'Sikorsky', 'SIKORSKY AIRCRAFT CORP': 'Sikorsky', 'Sikorsky/orlando': 'Sikorsky', 'SILVAIRE': 'Silvaire', 'SKYKITS': 'Skykits', 'SKYKITS CORP': 'Skykits', 'Skykits Corporation': 'Skykits', 'SKYKITS USA CORP': 'Skykits', 'SMITH': 'Smith', 'Smith & R. Mathews': 'Smith', 'Smith Aerostar': 'Smith', 'SMITH ALBERT F': 'Smith', 'SMITH ALLEN': 'Smith', 'Smith Arthur Fox': 'Smith', 'SMITH BRET B': 'Smith', 'SMITH Capella': 'Smith', 'Smith Carter A': 'Smith', 'Smith Douglas J.': 'Smith', 'SMITH EDWARD I': 'Smith', 'Smith Mini': 'Smith', 'Smith Miniplane': 'Smith', 'SMITH MINIPLANE': 'Smith', 'SMITH VILAS': 'Smith', 'Smith Wylie Jay': 'Smith', 'Smith, Ted Aerostar': 'Smith', 'Smith/davis': 'Smith', 'SNOW': 'Snow', 'SOCATA': 'Socata', 'Socata-Groupe Aerospatiale': 'Socata', 'SONEX': 'Sonex', 'Sonex / John D. McCarter': 'Sonex', 'SONEX AIRCRAFT': 'Sonex', 'SONEX LIMITED': 'Sonex', 'SORENSEN': 'Sorensen', 'SORENSEN DANNY': 'Sorensen', 'SORENSEN DANNY S': 'Sorensen', 'SORENSON': 'Sorensen', 'Sorrel': 'Sorrell', 'Sorrell Aircraft': 'Sorrell', 'STANLEY': 'Stanley', 'STANLEY ARTHUR FREEMAN': 'Stanley', 'STANLEY B E': 'Stanley', 'STANLEY ERNIE SIGURD': 'Stanley', 'Stanley, Davey L': 'Stanley', 'STANTON': 'Stanton', 'Star Duster': 'Starduster', 'Star Duster Too': 'Starduster', 'Starduster Ii': 'Starduster', 'STARDUSTER II': 'Starduster', 'Starduster Too': 'Starduster', 'STAUDACHER AIRCRAFT INC': 'Staudacher', 'Staudacher Aircraft, Inc.': 'Staudacher', 'STAUDACHER HYDROPLANES': 'Staudacher', 'STAUDACHER JON': 'Staudacher', 'Staudaucher': 'Staudacher', 'STEARMAN': 'Stearman', 'STEARMAN AIRCRAFT': 'Stearman', 'STEELE': 'Steele', 'STEELE  JOHN J': 'Steele', 'STEELE RALPH BRUCE': 'Steele', 'STEELE SAMUEL D': 'Steele', 'STEEN': 'Steen', 'Steen Aero Lab': 'Steen', 'Steen Skybolt': 'Steen', 'STINSON': 'Stinson', 'Stits Aircraft': 'Stits', 'Stits Flut-r-bug': 'Stits', 'STITS FLUT-R-BUG': 'Stits', 'Stits Playboy': 'Stits', 'Stits-itrich': 'Stits', 'Stitts': 'Stits', 'STODDARD HAMILTON': 'Stoddard Hamilton', 'Stoddard-Hamilton': 'Stoddard Hamilton', 'STOL': 'Stol', 'Stol Aircraft': 'Stol', 'STOL Aircraft Corp': 'Stol', 'STOL LLC': 'Stol', 'STOLP STARDUSTER': 'Stolp Starduster', 'Stolp Starduster Corp.': 'Stolp Starduster', 'Stolp-adams': 'Stolp Starduster', 'Stolp-starduster Too': 'Stolp Starduster', 'SUKHOI': 'Sukhoi', 'SUTTON': 'Sutton', 'Sutton Tailwind': 'Sutton', 'SUTTON WILLIAM J': 'Sutton', 'SWANSON': 'Swanson', 'Swanson/bensen': 'Swanson', 'SWEARINGEN': 'Swearingen', 'Swearingen T R/masters W': 'Swearingen', 'TAYLOR': 'Taylor', 'Taylor Air Command': 'Taylor', 'Taylor Lonsdale': 'Taylor', 'Taylor Smith': 'Taylor', 'Taylor Titch': 'Taylor', 'TAYLORCRAFT': 'Taylorcraft', 'Taylorcraft Aviation': 'Taylorcraft', 'TAYLORCRAFT AVIATION CORP': 'Taylorcraft', 'TAYLORCRAFT CORP': 'Taylorcraft', 'Taylorcraft Corporation': 'Taylorcraft', 'TECNAM': 'Tecnam', 'TEMCO': 'Temco', 'Temco Luscombe': 'Temco', 'TERATORN': 'Teratorn', 'Teratorn Acft Inc.': 'Teratorn', 'Teratorn Aircraft, Inc.': 'Teratorn', 'Teratron': 'Teratorn', 'TEXAS HELICOPTER CORP': 'Texas Helicopter', 'Texas Helicopter Corp.':'Texas Helicopter', 'Texas Helicopter Corporation': 'Texas Helicopter', 'TEXTRON AVIATION INC': 'Textron Aviation', 'Textron Aviation Inc': 'Textron Aviation', 'THE BOEING COMPANY': 'Boeing', 'THOMPSON': 'Thompson', 'THORP': 'Thorp', 'Thorp Aero, Inc.': 'Thorp', 'Thorpe': 'Thorp', 'THRUSH': 'Thrush', 'THRUSH AIRCRAFT INC': 'Thrush', 'Thrush Aircraft Inc.': 'Thrush', 'THRUSH AIRCRAFT LLC': 'Thrush', 'Thrush Aircraft, Inc.': 'Thrush', 'TITAN': 'Titan', 'TITAN AEROSPACE HOLDINGS INC': 'Titan', 'Titan Aircraft': 'Titan', 'TRAVEL AIR': 'Travel Air', 'TUPOLEV': 'Tupolev', 'Univair Aircraft Corporation': 'Univair', 'UNIVAIR AIRCRAFT CORPORATION': 'Univair', 'Univar': 'Univair', 'UNIVERSAL': 'Universal', 'Universal Aircraft Industries': 'Universal', 'Universal Globe': 'Universal', 'Universal Moulded Pdts.': 'Universal', 'Universal Stinson': 'Universal', 'UNIVERSAL STINSON': 'Universal', 'VANS': 'Vans', 'Vans Aircraft': 'Vans', 'Vans Aircraft Inc': 'Vans', 'VANS AIRCRAFT INC': 'Vans', 'Vans Aircraft, Inc.': 'Vans', 'VARGA AIRCRAFT CORP.': 'Varga', 'VELOCITY INC': 'Velocity', 'VICKERS': 'Vickers', 'WACO': 'Waco', 'Waco Classic Aircraft': 'Waco', 'WACO CLASSIC AIRCRAFT': 'Waco', 'Waco Classic Aircraft Corp': 'Waco', 'WACO CLASSIC AIRCRAFT CORP': 'Waco', 'Waco Classic Aircraft Corp.': 'Waco', 'WEATHERLY': 'Weatherly', 'WEATHERLY AVIATION CO INC': 'Weatherly', 'Weatherly Aviation Company Inc': 'Weatherly', 'WEBER': 'Weber', 'WHEELER': 'Wheeler', 'Wheeler Acft. Co.': 'Wheeler', 'WHEELER C / WHEELER K': 'Wheeler', 'Wheeler Technology, Inc.': 'Wheeler', 'WHITTMAN': 'Whittman', 'Whittman Tailwind': 'Whittman', 'WILSON': 'Wilson', 'Wing Aircraft': 'Wing', 'Wing Aircraft Co.': 'Wing', 'WOOD': 'Wood', 'WSK PZL MIELEC': 'Wsk Pzl Mielec', 'Wsk Pzl Swidnik': 'Wsk Pzl Mielec', 'Wsk Pzl Warzawa-okecie': 'Wsk Pzl Mielec', 'Wsk Pzl-krosno': 'Wsk Pzl Mielec', 'WSK-MIELEC': 'Wsk Pzl Mielec', 'WSK-PZL MEILEC': 'Wsk Pzl Mielec', 'Wsk-pzl Mielec': 'Wsk Pzl Mielec', 'WSK-PZL WARZAWA-OKECIE': 'Wsk Pzl Mielec', 'Wsk-pzl Warzawaokecie': 'Wsk Pzl Mielec', 'WSL PZL': 'Wsk Pzl Mielec', 'XTREMEAIR GMBH': 'Xtremeair GMBH', 'YAKOVLEV': 'Yakovlev', 'YAKOVLEV/CHINNERY': 'Yakovlev', 'YAKOVLEV/DAY': 'Yakovlev', 'ZENAIR': 'Zenair', 'ZENAIR LTD': 'Zenair', 'Zenair Zodiac': 'Zenair', 'ZENITH': 'Zenith', 'ZENITH ACFT CO': 'Zenith', 'ZENITH AIRCRAFT CO': 'Zenith', 'ZIMMERMAN': 'Zimmerman', 'ZIVKO AERONAUTICS INC': 'Zivko', 'Zivko Aeronautics': 'Zivko', 'Zivko Aeronautics Inc.': 'Zivko', 'ZLIN': 'Zlin', 'Zlin Aviation': 'Zlin', 'Zlin Aviation S.r.o.': 'Zlin'}

df['Make'] = df['Make'].replace(make_column_name_replace)

In [177]:
# Clean up make column for McDonnell Douglas Helicopters and Bell
make_column_name_replace = {'Md Helicopter': 'McDonnell Douglas Helicopters',
                            'Mcdonnell Douglas Helicopter':
                            'McDonnell Douglas Helicopters', 'McDonnell Douglas Helicopter':
                            'McDonnell Douglas Helicopters', 'BELL': 'Bell',
                            'BELL HELICOPTER TEXTRON CANADA': 'Bell',
                            'BELL HELICOPTER TEXTRON': 'Bell'}

df['Make'] = df['Make'].replace(make_column_name_replace)

In [178]:
# Show Make value_counts over 10
makes_value_10 = df['Make'].value_counts()[df['Make'].value_counts() > 10]
makes_value_10

Make
Cessna              26903
Piper               14818
Beech                5431
Bell                 2750
Boeing               2726
                    ...  
Univair                11
Bushby                 11
Eclipse Aviation       11
Hispano Aviacion       11
Flightstar             11
Name: count, Length: 235, dtype: int64

In [179]:
# Show the Aircraft_Category value_counts for makes_value_10 including NaN
df[df['Make'].isin(makes_value_10.index)]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
Airplane             42498
NaN                  32784
Helicopter            3164
Glider                 326
Balloon                128
Weight-Shift            60
Gyrocraft               14
Ultralight              10
Unknown                 10
Powered-Lift             4
WSFT                     4
Powered Parachute        1
Name: count, dtype: int64

In [180]:
# Show Aircraft Category value_counts for Cessna, Piper, Beech, Boeing, and Mooney, including Nan
df[df['Make'].isin(['Cessna', 'Piper', 'Beech', 'Boeing', 'Mooney'])][
    'Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
Airplane        35183
NaN             16062
Helicopter          2
Powered-Lift        1
Unknown             1
Name: count, dtype: int64

In [181]:
# Function to make all category entries for particular makes 'Airplane'
df.loc[df['Make'].isin(['Cessna', 'Piper', 'Beech', 'Boeing', 'Mooney']), 'Aircraft_Category'] = 'Airplane'

In [182]:
# Show Aircraft Category value_counts for Cessna, Piper, Beech, Boeing, and Mooney, including Nan
df[df['Make'].isin(['Cessna', 'Piper', 'Beech', 'Boeing', 'Mooney'])][
    'Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
Airplane    51249
Name: count, dtype: int64

In [183]:
# Show the Aircraft_Category value_counts for makes_value_10 including NaN
df[df['Make'].isin(makes_value_10.index)]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
Airplane             58564
NaN                  16722
Helicopter            3162
Glider                 326
Balloon                128
Weight-Shift            60
Gyrocraft               14
Ultralight              10
Unknown                  9
WSFT                     4
Powered-Lift             3
Powered Parachute        1
Name: count, dtype: int64

In [184]:
# Show Helicopter value_counts
df[df['Aircraft_Category'] == 'Helicopter']['Make'].value_counts()

Make
Robinson                981
Bell                    948
Hughes                  248
Eurocopter              190
Schweizer               115
                       ... 
BOYKIN STEPHEN VANCE      1
PIASECKI/PIKE             1
Embraer                   1
SMITH RICHARD D JR        1
CHILDS MICHAEL  A         1
Name: count, Length: 238, dtype: int64

In [185]:
# Function to make all category entries for particular makes 'Helicopter'
df.loc[df['Make'].isin(['Robinson', 'Bell', 'Hughes', 'Eurocopter', 'Schweizer']), 'Aircraft_Category'] = 'Helicopter'

In [186]:
# Show Helicopter value_counts
df[df['Aircraft_Category'] == 'Helicopter']['Make'].value_counts()

Make
Bell                    2750
Robinson                1672
Hughes                   935
Schweizer                800
Eurocopter               295
                        ... 
BOYKIN STEPHEN VANCE       1
PIASECKI/PIKE              1
Embraer                    1
SMITH RICHARD D JR         1
CHILDS MICHAEL  A          1
Name: count, Length: 238, dtype: int64

In [187]:
# Show the Aircraft_Category value_counts for makes_value_10 including NaN
df[df['Make'].isin(makes_value_10.index)]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
Airplane             58530
NaN                  12901
Helicopter            7132
Glider                 215
Balloon                128
Weight-Shift            60
Gyrocraft               14
Ultralight              10
Unknown                  5
WSFT                     4
Powered-Lift             3
Powered Parachute        1
Name: count, dtype: int64

In [188]:
# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
Grumman               1207
Bellanca               750
Air Tractor            462
Mcdonnell Douglas      453
Aeronca                399
                      ... 
Dewitt, Richard A.       1
Greth                    1
Robert D. Waldron        1
Alkire                   1
BELLANCA                 1
Name: count, Length: 3800, dtype: int64

In [189]:
# Show category value_counts for Grumman, Bellanca, Air Tractor, and Aeronca, including NaN
df[df['Make'].isin(['Grumman', 'Bellanca', 'Air Tractor', 'Aeronca'])][
    'Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN         2818
Airplane    1226
Name: count, dtype: int64

In [190]:
# Fill in 'Airplane' for Grumman, Bellanca, Air Tractor, and Aeronca
df.loc[df['Make'].isin(['Grumman', 'Bellanca', 'Air Tractor', 'Aeronca']), 'Aircraft_Category'] = 'Airplane'

In [191]:
# Show category value_counts for Grumman, Bellanca, Air Tractor, and Aeronca, including NaN
df[df['Make'].isin(['Grumman', 'Bellanca', 'Air Tractor', 'Aeronca'])][
    'Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
Airplane    4044
Name: count, dtype: int64

In [192]:
# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
Mcdonnell Douglas     453
Maule                 356
Champion              347
de Havilland          328
Aero Commander        318
                     ... 
Dewitt, Richard A.      1
Greth                   1
Robert D. Waldron       1
Alkire                  1
BELLANCA                1
Name: count, Length: 3796, dtype: int64

In [193]:
# Show category value_counts for Maule, Champion, de Havilland, Aero Commander, including NaN
df[df['Make'].isin(['Maule', 'Champion', 'de Havilland', 'Aero Commander'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN         1349
Airplane     694
Name: count, dtype: int64

In [194]:
# Fill in 'Airplane' for Maule, Champion, de Havilland, Aero Commander
df.loc[df['Make'].isin(['Maule', 'Champion', 'de Havilland', 'Aero Commander']), 'Aircraft_Category'] = 'Airplane'

In [195]:
# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
Mcdonnell Douglas     453
Rockwell              311
Hiller                290
Stinson               287
Aerospatiale          283
                     ... 
Dewitt, Richard A.      1
Greth                   1
Robert D. Waldron       1
Alkire                  1
BELLANCA                1
Name: count, Length: 3792, dtype: int64

In [196]:
# Show category value_counts for Rockwell, Hiller, Stinson, Aerospatiale, including NaN
df[df['Make'].isin(['Rockwell', 'Stinson'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN         598
Airplane    270
Name: count, dtype: int64

In [197]:
# Fill in 'Airplane' for Rockwell and Stinson
df.loc[df['Make'].isin(['Rockwell', 'Stinson']), 'Aircraft_Category'] = 'Airplane'

In [198]:
# Deal with Hiller and Aerospatiale
df[df['Make'].isin(['Aerospatiale', 'Hiller'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN                  573
Helicopter           163
Airplane               3
Powered Parachute      1
Name: count, dtype: int64

In [199]:
# Since Hiller and Aerospatiale are overwhelmly Helicopter, let's fill those in
df.loc[df['Make'].isin(['Aerospatiale', 'Hiller']), 'Aircraft_Category'] = 'Helicopter'

In [200]:
# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
Mcdonnell Douglas     453
Taylorcraft           269
North American        265
Luscombe              248
Douglas               222
                     ... 
Dewitt, Richard A.      1
Greth                   1
Robert D. Waldron       1
Alkire                  1
BELLANCA                1
Name: count, Length: 3788, dtype: int64

In [201]:
df[df['Make'].isin(['Mcdonnell Douglas'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN           453
Airplane      121
Helicopter     15
Name: count, dtype: int64

In [202]:
# If Make is Mcdonnell Douglas and category is Helicopter, change make to McDonnell Douglas Helicopters
df.loc[(df['Make'] == 'Mcdonnell Douglas') & (df['Aircraft_Category'] == 'Helicopter'), 'Make'] = 'McDonnell Douglas Helicopters'

In [203]:
df[df['Make'].isin(['Mcdonnell Douglas'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN         453
Airplane    121
Name: count, dtype: int64

In [204]:
df.loc[df['Make'].isin(['Mcdonnell Douglas']), 'Aircraft_Category'] = 'Airplane'

In [205]:
# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
Taylorcraft           269
North American        265
Luscombe              248
Douglas               222
Enstrom               212
                     ... 
Dewitt, Richard A.      1
Greth                   1
Robert D. Waldron       1
Alkire                  1
BELLANCA                1
Name: count, Length: 3787, dtype: int64

In [211]:
df[df['Make'].isin(['Taylorcraft', 'North American', 'Luscombe', 'Douglas'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN         1004
Airplane     462
Name: count, dtype: int64

In [213]:
df.loc[df['Make'].isin(['Taylorcraft', 'North American', 'Luscombe', 'Douglas']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
Enstrom              212
Ayres                181
Ercoupe              168
Airbus               140
Sikorsky             136
                    ... 
Greth                  1
Robert D. Waldron      1
Alkire                 1
Donald L. Betchan      1
BELLANCA               1
Name: count, Length: 3783, dtype: int64

In [216]:
df[df['Make'].isin(['Enstrom'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN           212
Helicopter     91
Name: count, dtype: int64

In [217]:
df.loc[df['Make'].isin(['Enstrom']), 'Aircraft_Category'] = 'Helicopter'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
Ayres                181
Ercoupe              168
Airbus               140
Sikorsky             136
Gulfstream           135
                    ... 
Greth                  1
Robert D. Waldron      1
Alkire                 1
Donald L. Betchan      1
BELLANCA               1
Name: count, Length: 3782, dtype: int64

In [221]:
df[df['Make'].isin(['Ayres', 'Ercoupe', 'Gulfstream'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN         484
Airplane    257
Name: count, dtype: int64

In [222]:
df.loc[df['Make'].isin(['Ayres', 'Ercoupe', 'Gulfstream']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
Airbus               140
Sikorsky             136
Fairchild            131
Pitts                125
Balloon Works        120
                    ... 
Greth                  1
Robert D. Waldron      1
Alkire                 1
Donald L. Betchan      1
BELLANCA               1
Name: count, Length: 3779, dtype: int64

In [223]:
df[df['Make'].isin(['Airbus'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
Airplane        284
NaN             140
Helicopter       20
Powered-Lift      1
Name: count, dtype: int64

In [224]:
#Deal with Airbus name for helicopters
df.loc[(df['Make'] == 'Airbus') & (df['Aircraft_Category'] == 'Helicopter'), 'Make'] = 'Airbus Helicopters'

In [225]:
df[df['Make'].isin(['Airbus'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
Airplane        284
NaN             140
Powered-Lift      1
Name: count, dtype: int64

In [226]:
# Fill in the rest for Airbus
df.loc[df['Make'].isin(['Airbus']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
Sikorsky             136
Fairchild            131
Pitts                125
Balloon Works        120
Swearingen           120
                    ... 
Greth                  1
Robert D. Waldron      1
Alkire                 1
Donald L. Betchan      1
BELLANCA               1
Name: count, Length: 3778, dtype: int64

In [232]:
df[df['Make'].isin(['Sikorsky'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN           136
Helicopter     95
Name: count, dtype: int64

In [233]:
# Fill in the rest for Sikorsky
df.loc[df['Make'].isin(['Sikorsky']), 'Aircraft_Category'] = 'Helicopter'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
Fairchild            131
Pitts                125
Balloon Works        120
Swearingen           120
Lake                 114
                    ... 
Greth                  1
Robert D. Waldron      1
Alkire                 1
Donald L. Betchan      1
BELLANCA               1
Name: count, Length: 3777, dtype: int64

In [237]:
df[df['Make'].isin(['Fairchild', 'Pitts', 'Swearingen', 'Lake'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN         490
Airplane    170
Name: count, dtype: int64

In [238]:
# Fill in the rest for previous
df.loc[df['Make'].isin(['Fairchild', 'Pitts', 'Swearingen', 'Lake']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
Balloon Works        120
Mitsubishi           109
Burkhart Grob        100
Let                   96
Waco                  95
                    ... 
Greth                  1
Robert D. Waldron      1
Alkire                 1
Donald L. Betchan      1
BELLANCA               1
Name: count, Length: 3773, dtype: int64

In [242]:
df[df['Make'].isin(['Mitsubishi', 'Waco'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN         204
Airplane     89
Unknown       1
Name: count, dtype: int64

In [243]:
# Fill in the rest for previous
df.loc[df['Make'].isin(['Mitsubishi', 'Waco']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
Balloon Works        120
Burkhart Grob        100
Let                   96
Lockheed              94
Ryan                  88
                    ... 
Greth                  1
Robert D. Waldron      1
Alkire                 1
Donald L. Betchan      1
BELLANCA               1
Name: count, Length: 3771, dtype: int64

In [249]:
df[df['Make'].isin(['Ryan', 'Lockheed'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN             182
Airplane         63
Powered-Lift      1
Name: count, dtype: int64

In [250]:
# Fill in the rest for previous
df.loc[df['Make'].isin(['Ryan', 'Lockheed']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
Balloon Works        120
Burkhart Grob        100
Let                   96
Aerostar              87
Learjet               86
                    ... 
Greth                  1
Robert D. Waldron      1
Alkire                 1
Donald L. Betchan      1
BELLANCA               1
Name: count, Length: 3769, dtype: int64

In [252]:
df[df['Make'].isin(['Learjet'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN         86
Airplane    59
Name: count, dtype: int64

In [253]:
# Fill in the rest for previous
df.loc[df['Make'].isin(['Balloon Works', 'Aerostar']), 'Aircraft_Category'] = 'Balloon'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
Burkhart Grob        100
Let                   96
Learjet               86
Helio                 85
Smith                 83
                    ... 
Greth                  1
Robert D. Waldron      1
Alkire                 1
Donald L. Betchan      1
BELLANCA               1
Name: count, Length: 3767, dtype: int64

In [254]:
# Fill in the rest for previous
df.loc[df['Make'].isin(['Burkhart Grob', 'Let']), 'Aircraft_Category'] = 'Glider'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
Learjet              86
Helio                85
Smith                83
Embraer              81
Raven                79
                     ..
Greth                 1
Robert D. Waldron     1
Alkire                1
Donald L. Betchan     1
BELLANCA              1
Name: count, Length: 3765, dtype: int64

In [258]:
df[df['Make'].isin(['Raven'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN        79
Balloon     9
Name: count, dtype: int64

In [259]:
# Fill in the rest for previous
df.loc[df['Make'].isin(['Learjet', 'Helio', 'Smith', 'Embraer']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
Raven                79
Wsk Pzl Mielec       79
British Aerospace    79
American Aviation    72
Aviat                72
                     ..
Greth                 1
Robert D. Waldron     1
Alkire                1
Donald L. Betchan     1
BELLANCA              1
Name: count, Length: 3761, dtype: int64

In [263]:
df[df['Make'].isin(['Aviat'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
Airplane    159
NaN          72
Name: count, dtype: int64

In [264]:
# Fill in the rest for previous
df.loc[df['Make'].isin(['Wsk Pzl Mielec', 'British Aerospace', 'American Aviation', 'Aviat']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
Raven                      79
Globe                      71
Weatherly                  69
Schleicher                 69
Cirrus                     65
                           ..
Darst                       1
Aero Falcon Intl., Inc.     1
Brauch                      1
Love, Sanford W.            1
BELLANCA                    1
Name: count, Length: 3757, dtype: int64

In [268]:
df[df['Make'].isin(['Cirrus'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
Airplane    401
NaN          65
Name: count, dtype: int64

In [269]:
# Fill in the rest for previous
df.loc[df['Make'].isin(['Globe', 'Weatherly', 'Cirrus']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
Raven                79
Schleicher           69
Mbb                  62
Gates Learjet        58
Schempp-hirth        57
                     ..
Greth                 1
Robert D. Waldron     1
Alkire                1
Donald L. Betchan     1
BELLANCA              1
Name: count, Length: 3754, dtype: int64

In [270]:
# Fill in the rest for previous
df.loc[df['Make'].isin(['Raven']), 'Aircraft_Category'] = 'Balloon'

In [272]:
# Fill in the rest for previous
df.loc[df['Make'].isin(['Schleicher']), 'Aircraft_Category'] = 'Glider'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
Mbb                  62
Gates Learjet        58
Schempp-hirth        57
Saab                 55
Navion               53
                     ..
Greth                 1
Robert D. Waldron     1
Alkire                1
Donald L. Betchan     1
BELLANCA              1
Name: count, Length: 3752, dtype: int64

In [277]:
df[df['Make'].isin(['Navion'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN         53
Airplane    26
Name: count, dtype: int64

In [278]:
# Fill in the rest for previous
df.loc[df['Make'].isin(['Mbb']), 'Aircraft_Category'] = 'Helicopter'
df.loc[df['Make'].isin(['Schempp-hirth']), 'Aircraft_Category'] = 'Glider'
df.loc[df['Make'].isin(['Gates Learjet', 'Saab', 'Navion']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
Canadair             53
                     51
Dassault             50
Socata               50
Cameron              47
                     ..
Greth                 1
Robert D. Waldron     1
Alkire                1
Donald L. Betchan     1
BELLANCA              1
Name: count, Length: 3747, dtype: int64

In [282]:
df[df['Make'].isin(['Cameron'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN        47
Balloon     7
Name: count, dtype: int64

In [283]:
df.loc[df['Make'].isin(['Balloon Works', 'Aerostar', 'Raven', 'Cameron']), 'Aircraft_Category'] = 'Balloon'
df.loc[df['Make'].isin(['Burkhart Grob', 'Let', 'Schleicher', 'Schempp-hirth']), 'Aircraft_Category'] = 'Glider'
df.loc[df['Make'].isin(['Robinson', 'Bell', 'Hughes', 'Eurocopter', 'Schweizer', 'Aerospatiale', 'Hiller', 'Enstrom', 'Sikorsky', 'Mbb']), 'Aircraft_Category'] = 'Helicopter'
df.loc[df['Make'].isin(['Cessna', 'Piper', 'Beech', 'Boeing', 'Mooney', 'Grumman', 'Bellanca', 'Air Tractor', 'Aeronca', 'Maule', 'Champion', 'de Havilland', 'Aero Commander', 'Rockwell', 'Stinson', 'Mcdonnell Douglas', 'Taylorcraft', 'North American', 'Luscombe', 'Douglas', 'Ayres', 'Ercoupe', 'Gulfstream', 'Airbus', 'Fairchild', 'Pitts', 'Swearingen', 'Lake', 'Mitsubishi', 'Waco', 'Ryan', 'Lockheed', 'Learjet', 'Helio', 'Smith', 'Embraer', 'Wsk Pzl Mielec', 'British Aerospace', 'American Aviation', 'Aviat', 'Globe', 'Weatherly', 'Cirrus', 'Gates Learjet', 'Saab', 'Navion', 'Canadair', 'Dassault', 'Socata']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
                       51
Fokker                 47
Bombardier             45
Rotorway               45
Christen Industries    45
                       ..
Greth                   1
Robert D. Waldron       1
Alkire                  1
Donald L. Betchan       1
BELLANCA                1
Name: count, Length: 3743, dtype: int64

In [287]:
df[df['Make'].isin(['Christen Industries'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN         45
Airplane    25
Name: count, dtype: int64

In [288]:
df.loc[df['Make'].isin(['Rotorway']), 'Aircraft_Category'] = 'Helicopter'
df.loc[df['Make'].isin(['Fokker', 'Bombardier', 'Christen Industries']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
                     51
Great Lakes          42
Eagle                42
Eipper               42
Convair              40
                     ..
Greth                 1
Robert D. Waldron     1
Alkire                1
Donald L. Betchan     1
BELLANCA              1
Name: count, Length: 3739, dtype: int64

In [292]:
df[df['Make'].isin(['Convair'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN         40
Airplane     9
Name: count, dtype: int64

In [293]:
df.loc[df['Make'].isin(['Great Lakes', 'Eagle', 'Eipper', 'Convair']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
                      51
Consolidated Aero     37
PZL                   36
Short                 35
Brantly Helicopter    35
                      ..
Greth                  1
Robert D. Waldron      1
Alkire                 1
Donald L. Betchan      1
BELLANCA               1
Name: count, Length: 3735, dtype: int64

In [297]:
df[df['Make'].isin(['Brantly Helicopter'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN           35
Helicopter     4
Name: count, dtype: int64

In [298]:
df.loc[df['Make'].isin(['Brantly Helicopter']), 'Aircraft_Category'] = 'Helicopter'
df.loc[df['Make'].isin(['Consolidated Aero', 'Short']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
                      51
PZL                   36
Aerotek               35
Kaman                 33
Rolladen Schneider    33
                      ..
Greth                  1
Robert D. Waldron      1
Alkire                 1
Donald L. Betchan      1
BELLANCA               1
Name: count, Length: 3732, dtype: int64

In [301]:
df[df['Make'].isin(['Rolladen Schneider'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN         33
Glider      14
Airplane     1
Name: count, dtype: int64

In [302]:
df.loc[df['Make'].isin(['Rolladen Schneider']), 'Aircraft_Category'] = 'Glider'
df.loc[df['Make'].isin(['Kaman']), 'Aircraft_Category'] = 'Helicopter'
df.loc[df['Make'].isin(['Aerotek']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
                     51
PZL                  36
Fairchild Hiller     33
Britten Norman       33
Raytheon             32
                     ..
Greth                 1
Robert D. Waldron     1
Alkire                1
Donald L. Betchan     1
BELLANCA              1
Name: count, Length: 3729, dtype: int64

In [305]:
df[df['Make'].isin(['Raytheon'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
Airplane    88
NaN         32
Name: count, dtype: int64

In [306]:
df.loc[df['Make'].isin(['Fairchild Hiller']), 'Aircraft_Category'] = 'Helicopter'
df.loc[df['Make'].isin(['Britten Norman', 'Raytheon']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
                     51
PZL                  36
Pilatus              32
Agusta               31
Diamond              30
                     ..
Greth                 1
Robert D. Waldron     1
Alkire                1
Donald L. Betchan     1
BELLANCA              1
Name: count, Length: 3726, dtype: int64

In [309]:
df[df['Make'].isin(['Diamond'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
Airplane        76
NaN             30
Powered-Lift     1
Glider           1
Name: count, dtype: int64

In [310]:
df.loc[df['Make'].isin(['Agusta']), 'Aircraft_Category'] = 'Helicopter'
df.loc[df['Make'].isin(['Pilatus', 'Diamond']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
                       51
PZL                    36
Alon                   29
Hawker                 28
Continental Copters    28
                       ..
Greth                   1
Robert D. Waldron       1
Alkire                  1
Donald L. Betchan       1
BELLANCA                1
Name: count, Length: 3723, dtype: int64

In [313]:
df[df['Make'].isin(['Continental Copters'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN    28
Name: count, dtype: int64

In [314]:
df.loc[df['Make'].isin(['Continental Copters']), 'Aircraft_Category'] = 'Helicopter'
df.loc[df['Make'].isin(['Alon', 'Hawker']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
                     51
PZL                  36
Texas Helicopter     28
American Champion    27
Republic             27
                     ..
Greth                 1
Robert D. Waldron     1
Alkire                1
Donald L. Betchan     1
BELLANCA              1
Name: count, Length: 3720, dtype: int64

In [317]:
df[df['Make'].isin(['Republic'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN         27
Airplane     9
Name: count, dtype: int64

In [318]:
df.loc[df['Make'].isin(['Texas Helicopter']), 'Aircraft_Category'] = 'Helicopter'
df.loc[df['Make'].isin(['American Champion', 'Republic']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
                     51
PZL                  36
Homebuilt            27
Rans                 27
Dornier              25
                     ..
Greth                 1
Robert D. Waldron     1
Alkire                1
Donald L. Betchan     1
BELLANCA              1
Name: count, Length: 3717, dtype: int64

In [321]:
df[df['Make'].isin(['Dornier'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN         25
Airplane     9
Name: count, dtype: int64

In [322]:
df.loc[df['Make'].isin(['Homebuilt', 'Rans', 'Dornier']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
                              51
PZL                           36
I.c.a. Brasov                 25
Siai Marchetti                25
Israel Aircraft Industries    24
                              ..
Greth                          1
Robert D. Waldron              1
Alkire                         1
Donald L. Betchan              1
BELLANCA                       1
Name: count, Length: 3714, dtype: int64

In [325]:
df[df['Make'].isin(['Israel Aircraft Industries'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN         24
Airplane    13
Name: count, dtype: int64

In [326]:
df.loc[df['Make'].isin(['I.c.a. Brasov']), 'Aircraft_Category'] = 'Glider'
df.loc[df['Make'].isin(['Siai Marchetti', 'Israel Aircraft Industries']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
                     51
PZL                  36
Snow                 23
Yakovlev             23
Quicksilver          22
                     ..
Greth                 1
Robert D. Waldron     1
Alkire                1
Donald L. Betchan     1
BELLANCA              1
Name: count, Length: 3711, dtype: int64

In [329]:
df[df['Make'].isin(['Quicksilver'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
Airplane      53
NaN           22
Ultralight     3
Unknown        1
Name: count, dtype: int64

In [330]:
df.loc[df['Make'].isin(['Snow', 'Yakovlev', 'Quicksilver']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
                     51
PZL                  36
Adams Balloon        22
Lancair              22
Callair              21
                     ..
Greth                 1
Robert D. Waldron     1
Alkire                1
Donald L. Betchan     1
BELLANCA              1
Name: count, Length: 3708, dtype: int64

In [333]:
df[df['Make'].isin(['Callair'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN         21
Airplane     5
Name: count, dtype: int64

In [334]:
df.loc[df['Make'].isin(['Adams Balloon']), 'Aircraft_Category'] = 'Balloon'
df.loc[df['Make'].isin(['Lancair', 'Callair']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
                     51
PZL                  36
Varga                21
Thunder And Colt     21
Quickie              21
                     ..
Greth                 1
Robert D. Waldron     1
Alkire                1
Donald L. Betchan     1
BELLANCA              1
Name: count, Length: 3705, dtype: int64

In [337]:
df[df['Make'].isin(['Quickie'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN         21
Airplane    14
Name: count, dtype: int64

In [338]:
df.loc[df['Make'].isin(['Thunder And Colt']), 'Aircraft_Category'] = 'Balloon'
df.loc[df['Make'].isin(['Varga', 'Quickie']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
                     51
PZL                  36
Garlick              21
Extra                20
Glasflugel           19
                     ..
Greth                 1
Robert D. Waldron     1
Alkire                1
Donald L. Betchan     1
BELLANCA              1
Name: count, Length: 3702, dtype: int64

In [343]:
# Show Make values whose Aircraft_Category value is NaN if there are over 15
df[df['Aircraft_Category'].isna()]['Make'].value_counts()[df[df['Aircraft_Category'].isna()]['Make'].value_counts() > 15]

Make
                         51
PZL                      36
Garlick                  21
Extra                    20
Glasflugel               19
Curtiss Wright           18
Kolb                     18
Glasair                  18
ATR                      18
Casa                     18
Barnes                   17
Temco                    17
Johnson                  17
Classic Aircraft Corp    17
Unknown                  17
Air Command              17
Davis                    17
Miller                   16
Forney                   16
Glaser-dirks             16
Name: count, dtype: int64

In [360]:
df[df['Make'].isin(['Glaser-dirks'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN       16
Glider     1
Name: count, dtype: int64

In [361]:
df.loc[df['Make'].isin(['Barnes']), 'Aircraft_Category'] = 'Balloon'
df.loc[df['Make'].isin(['Glasflugel', 'Glaser-dirks']), 'Aircraft_Category'] = 'Glider'
df.loc[df['Make'].isin(['Garlick']), 'Aircraft_Category'] = 'Helicopter'
df.loc[df['Make'].isin(['Extra', 'Curtiss Wright', 'Kolb', 'Glasair', 'ATR', 'Casa', 'Temco', 'Johnson', 'Classic Aircraft Corp', 'Davis', 'Miller', 'Forney']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
                     51
PZL                  36
Air Command          17
Unknown              17
Interstate           15
                     ..
Greth                 1
Robert D. Waldron     1
Alkire                1
Donald L. Betchan     1
BELLANCA              1
Name: count, Length: 3686, dtype: int64

In [362]:
# Show Make values whose Aircraft_Category value is NaN if there are over 10
df[df['Aircraft_Category'].isna()]['Make'].value_counts()[df[df['Aircraft_Category'].isna()]['Make'].value_counts() > 10]

Make
                             51
PZL                          36
Air Command                  17
Unknown                      17
Interstate                   15
Eiriavion Oy                 15
Sukhoi                       15
Artic Aircraft Corp.         15
Vans                         15
Rotec                        15
Thorp                        15
Anderson Aircraft Corp.      15
American General Aircraft    15
Culver                       14
Mitchell                     13
Bensen                       13
Stearman                     13
Pterodactyl                  12
Aerofab Inc.                 12
Hall                         12
Taylor                       12
Weedhopper                   12
Air & Space                  12
Nord                         11
Jones                        11
Hispano Aviacion             11
Young                        11
Rutan                        11
Naval Aircraft Factory       11
Howard Aircraft              11
Steen                        11
Teratorn                     11
Meyers                       11
Starduster                   11
Partenavia                   11
Name: count, dtype: int64

In [393]:
df[df['Make'].isin(['Partenavia'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN         11
Airplane     1
Name: count, dtype: int64

In [395]:
df.loc[df['Make'].isin(['Bensen', 'Air & Space']), 'Aircraft_Category'] = 'Gyrocraft'
df.loc[df['Make'].isin(['Pterodactyl', 'Weedhopper']), 'Aircraft_Category'] = 'Ultralight'
df.loc[df['Make'].isin(['Eiriavion Oy']), 'Aircraft_Category'] = 'Glider'
df.loc[df['Make'].isin(['Interstate', 'Sukhoi', 'Artic Aircraft Corp.', 'Vans', 'Rotec', 'Thorp', 'Anderson Aircraft Corp.', 'American General Aircraft', 'Culver', 'Mitchell', 'Stearman', 'Aerofab Inc.', 'Hall', 'Taylor', 'Nord', 'Jones', 'Hispano Aviacion', 'Young', 'Rutan', 'Naval Aircraft Factory', 'Howard Aircraft', 'Steen', 'Teratorn', 'Meyers', 'Starduster', 'Partenavia']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
                     51
PZL                  36
Air Command          17
Unknown              17
Maxair               10
                     ..
Greth                 1
Robert D. Waldron     1
Alkire                1
Donald L. Betchan     1
BELLANCA              1
Name: count, Length: 3655, dtype: int64

In [406]:
# Show Make values whose Aircraft_Category value is NaN if there are over 5 and under 11
blanks_over_5 = df[df['Aircraft_Category'].isna()]['Make'].value_counts()[(df[df['Aircraft_Category'].isna()]['Make'].value_counts() > 5)]

# Show blank_over_5 below 11
blanks_over_5[blanks_over_5 < 11]

Make
Maxair                            10
Benson                            10
Bede Aircraft                     10
American Aerolights               10
Martin                            10
                                  ..
Harris                             6
Government Aircraft Fact (gaf)     6
Experimental                       6
Stevens                            6
Hayes                              6
Name: count, Length: 77, dtype: int64

In [416]:
df[df['Make'].isin(['Hayes'])]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
NaN    6
Name: count, dtype: int64

In [417]:
df.loc[df['Make'].isin(['Benson']), 'Aircraft_Category'] = 'Gyrocraft'
df.loc[df['Make'].isin(['American Aerolights']), 'Aircraft_Category'] = 'Ultralight'
df.loc[df['Make'].isin(['Maxair', 'Bede Aircraft', 'Martin']), 'Aircraft_Category'] = 'Airplane'

# Show Make values whose Aircraft_Category value is NaN
df[df['Aircraft_Category'].isna()]['Make'].value_counts()

Make
                     51
PZL                  36
Unknown              17
Air Command          17
Nanchang             10
                     ..
Greth                 1
Robert D. Waldron     1
Alkire                1
Donald L. Betchan     1
BELLANCA              1
Name: count, Length: 3650, dtype: int64

In [418]:
# Show Make value_counts over 10
makes_value_10 = df['Make'].value_counts()[df['Make'].value_counts() > 10]
makes_value_10

Make
Cessna              26903
Piper               14818
Beech                5431
Bell                 2750
Boeing               2726
                    ...  
BURKHART GROB          11
Univair                11
Bushby                 11
Hispano Aviacion       11
Blanik                 11
Name: count, Length: 235, dtype: int64

In [419]:
# Show the Aircraft_Category value_counts for makes_value_10 including NaN
df[df['Make'].isin(makes_value_10.index)]['Aircraft_Category'].value_counts(dropna=False)

Aircraft_Category
Airplane        68918
Helicopter       8374
Glider            645
Balloon           521
NaN               402
Weight-Shift       60
Gyrocraft          49
Ultralight         29
WSFT                4
Unknown             1
Name: count, dtype: int64